### PR TITLE
Bonnie 1203

### PR DESCRIPTION
**Pull request by jbradl11**
Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

This PR is to address the lack of tests for Bonnie Rake tasks and to remove Rake tasks that no longer provide value.

JIRA Ticket: https://jira.mitre.org/browse/BONNIE-1203

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR:
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * See [front-end testing instructions](https://github.com/projecttacoma/bonnie/wiki/Testing#frontend-testing) for getting coverage for front-end tests
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] N/A JIRA test links:
- [x] N/A Justification for using JIRA tests:
- [x] N/A JIRA tests have been added to sprint

**Reviewer 1:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @mayerm94
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: test/rake/bonnie_users_test.rb [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    assert_equal(1, dest_measures.count)
+    assert_equal(measure._id, dest_measures.first._id)
+
+    assert_equal(0, source_patients.count)
+    assert_equal(7, dest_patients.count)
+
+    assert_equal(vs_count, HealthDataStandards::SVS::ValueSet.where(user_id: source_user).count)
+    assert_equal(vs_count, HealthDataStandards::SVS::ValueSet.where(user_id: dest_user).count)
+  end
+
```
Comment (lukoktonos) [inline test/rake/bonnie_users_test.rb:207]: Will need to migrate the new test in https://github.com/projecttacoma/bonnie/pull/914 to this file.

File: lib/tasks/cql_testing.rake [hunk 2]
```diff
@@ -228,6 +228,139 @@ namespace :bonnie do
...
+            target_path = path.basename
+          end
+          puts "  + #{target_path}"
+          new_package_zip.add(target_path, path)
+        end
+      end
+    end
   end
-end
+end
```
Comment (lukoktonos) [inline lib/tasks/cql_testing.rake]: Should put the newline at the end back in here

File: test/rake/bonnie_patients_test.rb [hunk 3]
```diff
@@ -310,4 +310,94 @@ class RakeTest < ActiveSupport::TestCase
...
+    ENV['YEARS'] = '1'
+    ENV['DIR'] = 'forward'
+    p = Record.where(user_id:@source_user.id).first
+    before_birth_date = p.birthdate
+    before_dc_start_date = p.source_data_criteria[0]['start_date']
+    before_dc_end_date = p.source_data_criteria[0]['end_date']
+    Rake::Task['bonnie:patients:date_shift'].execute
+    p = Record.where(user_id:@source_user.id).first
+
+    # Aseert the before date + a year, equals the new birthdate of the patient and the new start and end dates of the data criteria.
```
Comment (lukoktonos) [inline test/rake/bonnie_patients_test.rb]: Aseert -> Assert

File: test/rake/bonnie_helpers_test.rb [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    assert_equal(6, value_sets_dest.count)
+    value_sets_dest.each do |vs|
+      assert_equal(@dest_user, vs.user)
+      assert_equal(@dest_user.bundle, vs.bundle)
+    end
+
+    # test transfer of value sets
+
+    copy_value_sets(@dest_user, value_sets_src)
```
Comment (lukoktonos) [inline test/rake/bonnie_helpers_test.rb:113]: It would be good to have some of the valuesets in `value_sets_hash_dest` not overlap with `value_sets_hash_src` in order to make sure that the rake task isn't clobbering any values.

**Review by lukoktonos — CHANGES_REQUESTED** [review]

File: test/rake/bonnie_patients_test.rb [hunk 5]
```diff
@@ -310,4 +310,94 @@ class RakeTest < ActiveSupport::TestCase
...
+    hqmf_set_id =  '5375D6A9-203B-4FFF-B851-AFA9B68D2AC2'
+    associate_user_with_measures(@source_user, CqlMeasure.where(hqmf_set_id: hqmf_set_id))
+
+    ENV['EMAIL'] = @source_user.email
+    ENV['HQMF_SET_ID'] = hqmf_set_id
+    ENV['FILENAME'] = File.join('test','fixtures','patient_export','cms104v2_export_patients_test.json')
+    ENV['MEASURE_TYPE'] = 'CQL'
+    Rake::Task['bonnie:patients:import_patients'].execute
+    # Confirm that there are 7 patients now associated with this measure.
+    assert_equal 7, Record.where(measure_ids: hqmf_set_id).count
```
Comment (mayerm94) [inline test/rake/bonnie_patients_test.rb]: This query should be for the hqmf_set_id of the measure AND user_id

File: test/rake/bonnie_helpers_test.rb (hunk 4 above)
Comment (edeyoung) [reply test/rake/bonnie_helpers_test.rb:113]: addressed

File: test/rake/bonnie_users_test.rb (hunk 1 above)
Comment (edeyoung) [reply test/rake/bonnie_users_test.rb:207]: addressed (it's in bonnie_patients_test. i didn't do this move - someone else did.)

**Review by lukoktonos — DISMISSED** [review]

**Review by lukoktonos — APPROVED** [review]

**Review by mayerm94 — APPROVED** [review]
Approve of commits made by Lizzie since my last review. Branch update required
